src/Integration/State/StateManager.cs

```diff
@@ -39,6 +40,7 @@ public StateManager(IHost host, TransferableVisualState state)
 
             this.Host = host;
             this.ManagedState = state;
+            this.isSolutionBound = this.ManagedState.HasBoundProject;
```

Comment:
Yes and no. 
Yes - we do want to delay the initialization, and no you can't set the correct value.
The reason being - the state manager manages the visual state of TE section.
By the sound of it you need two things
1) The change event that you previously added (binding is a logical operation driven by the UI, which is why it's fine that the state manager exposes it).
2) A service to get the initial state of the solution (bound or not) - I have a pending PR with such service (ISolutionBindingInformationProvider)

At the moment, once you have the initial state [ISolutionBindingInformationProvider.GetBoundProjects().Any()] you will need to update the IsActiveSolutionBound flag when the solution changes (IActiveSolutionTracker.ActiveSolutionChanged) or in response to IStateManager.BindingStateChanged event.